### ISSUE TITLE
[feature-request] - option to generate NFO and other metadata about the downloaded file

### ISSUE THREAD

**Issue by IlmariKu** [aajanki/yle-dl#361]
I'm using Jellyfin to organize my media and when I'm using Yle-dl to watch the content, Jellyfin is not recognizing the video files or doesn't know anything other about them, than the filename. Areena does contain descriptions and lot of other info that would be needed to fill the .nfo-file.

NFO-files are just xml in a specific format, saved next to the file.

Can or could `yle-dl` be adapted to save the description, name of the film and other metadata present in the movie page? I see there is an option for the metadata-json, so some info is already gathered there.

**Comment by IlmariKu** [comment]
I was thinking I could do make a PR about this myself, but was raising it as a question / idea to see if anyone's got something to say about it.

**Comment by aajanki** [comment]
Sounds like a nice feature! I'm happy to review a PR if you want to implement this.

There should be a command line flag for enabling/disabling the creation of NFO files for those who do not use them.

**Comment by IlmariKu** [comment]
Yep, definitely only an optional command-line flag, never the default, since additional operations are created.

I'll create the feature at some timescale over the summer and send a PR over. If anyone wants to collaborate on it, let me know!

**Comment by IlmariKu** [comment]
What should the flag be called? I was thinking, for Jellyfin / Kodi-usage, the feature should:
1) Create a folder (because metadata requires multiple files)
2) Create the .nfo-file and grab the cover-image
3) Extract subs from the stream (JF doesn't work well with MKV-embedded subs). This should probably be a standalone command as well.

Should it be multiple flags, like `--folderize(?)`, `--create-nfo`, `--extract-subs`, `--extract-cover` or just singular `--with-metadata` / `--jellyfin`, which would do all of those ops at once?

**Comment by aajanki** [comment]
For the user it would be easiest to have a single command line flag that enables everything. It could be called `--output-metadata-nfo` or something like that. Preferable not `--jellyfin` because NFO files would be usable with other software, too.

Is it necessary or a common practice to create a separate folder? Isn't it enough if the base filename is the same for the video, NFO and other files (e.g. kummeli_S01E01.mkv, kummeli_S01E01.nfo, kummeli_S01E01.srt)? Then different episodes or programs could be stored to the same folder.

**Comment by IlmariKu** [comment]
It's fine to not have a folder and for a playlist, that could be better, but individual movies stored under same root would benefit from it. I'll think about it when I'll start implementing this one day.

**Comment by aajanki** [comment]
Folder creation doesn't necessarily need a new command line argument. It is already possible with the existing `--output-template` argument, although the syntax is a bit verbose.

```
yle-dl --output-template '${series}_${episode}/${series_separator}${title}: ${episode_separator}${timestamp}'
```

This will save the video in a folder named with the series name and the episode number. The file name will consist of the series name, title, episode number and timestamp (the default output filename for yle-dl).

It's possible to enable this permanently in `~/.config/yledl.conf` instead of having to type it out every time.
